microceph/api/ops_replication.go

```diff
@@ -0,0 +1,89 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/canonical/microceph/microceph/api/types"
+	"github.com/canonical/microceph/microceph/ceph"
+	"github.com/canonical/microceph/microceph/interfaces"
+	"github.com/canonical/microcluster/v2/rest"
+	"github.com/canonical/microcluster/v2/state"
+)
+
+// Top level ops API
+var opsCmd = rest.Endpoint{
+	Path: "ops",
+}
+
+// replication ops API
+var opsReplicationCmd = rest.Endpoint{
+	Path: "ops/replication/",
+}
+
+// List Replications
+var opsReplicationWorkloadCmd = rest.Endpoint{
+	Path: "ops/replication/{wl}",
+	Get:  rest.EndpointAction{Handler: cmdOpsReplication, ProxyTarget: false},
+}
+
+// CRUD Replication
+var opsReplicationResourceCmd = rest.Endpoint{
+	Path:   "ops/replication/{wl}/{name}",
+	Get:    rest.EndpointAction{Handler: cmdOpsReplication, ProxyTarget: false},
+	Put:    rest.EndpointAction{Handler: cmdOpsReplication, ProxyTarget: false},
+	Delete: rest.EndpointAction{Handler: cmdOpsReplication, ProxyTarget: false},
+}
+
+// cmdOpsReplication is the common handler for all requests on replication endpoint.
+func cmdOpsReplication(s state.State, r *http.Request) response.Response {
+	// NOTE (utkarshbhatthere): unescaping API $wl and $name is not required
+	// as that information is present in payload.
```

Comment:
I'm -1 on taking $wl and $name from the passed in request data and ignoring the path variables. For a RESTful api it's important that the entities (here: replications) are indeed addressable via URI; by ignoring the path variable we'd run counter to that

Reply:
got it, fixed as per discussions.